WalletWasabi/WabiSabi/Backend/Rounds/CoinJoinStorage/CoinJoinTransactionArchiver.cs

```diff
@@ -50,6 +53,50 @@ public async Task<string> StoreJsonAsync(Transaction transaction, DateTimeOffset
 		return filePath;
 	}
 
+	public IEnumerable<TransactionInfo> ReadJson(DateTimeOffset from, DateTimeOffset to)
+	{
+		if (to < from)
+		{
+			throw new ArgumentException("Invalid time period", nameof(from));
+		}
+
+		List<string> dirsToRead = new();
+		int monthOffset = 0;
+
+		do
+		{
+			DateTimeOffset date = new(from.Year, from.Month + monthOffset, from.Day, 0, 0, 0, from.Offset);
+			if (date > to)
+			{
+				break;
+			}
+			dirsToRead.Add(Path.Combine(BaseStoragePath, $"{date:yyyy-MM}"));
+			monthOffset++;
+		}
+		while (true);
```

Comment:
There is also new [DateOnly API](https://www.stevejgordon.co.uk/using-dateonly-and-timeonly-in-dotnet-6) in .NET 6 which allows one to write something like this:

```cs
DateOnly dateOnly = DateOnly.FromDateTime(from.UtcDateTime);
// ...
dateOnly = dateOnly.AddMonths(1);
```

Maybe it would be a good alternative too.